src/components/directory-content.js

```diff
@@ -673,6 +681,45 @@ const DirectoryContent = () => {
         );
     };
 
+    const sort = useCallback(
+        (datakey, isAsc, isNumeric) => {
+            const rowsCopy = [...rows].map((row, index) => [row, index]);
+            if (datakey === 'type') {
+                rowsCopy.sort((rowIndex1, rowIndex2) => {
+                    const metadata1 =
+                        childrenMetadata[rowIndex1[0].elementUuid];
+                    const metadata2 =
+                        childrenMetadata[rowIndex2[0].elementUuid];
+                    return metadata1 && metadata2
+                        ? getElementTypeTranslation(
+                              rowIndex1[0].type,
+                              metadata1.subtype,
+                              metadata1.format
+                          ).localeCompare(
+                              getElementTypeTranslation(
+                                  rowIndex2[0].type,
+                                  metadata2.subtype,
+                                  metadata2.format
+                              )
+                          )
+                        : rowIndex1[0].type.localeCompare(rowIndex2[0].type);
+                });
+            } else if (isNumeric) {
```

Comment:
Maybe add a comment here to document that we customize sort method to manage `types` and `subtypes` sorting but we need to re-implement sort for standard cases for numerical an string values.